internal/controllers/node_label_module_version_reconciler_test.go

```diff
@@ -167,17 +167,44 @@ var _ = Describe("getDevicePluginPods", func() {
 	ctx := context.Background()
 
 	It("good flow", func() {
+		pod1 := v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{constants.ModuleNameLabel: "module name"},
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Kind: "some kind",
+					},
+					{
+						Kind: "DaemonSet",
+					},
+				},
+			},
+		}
+		pod2 := v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{constants.ModuleNameLabel: "other module name"},
+			},
+		}
+
 		fieldSelector := client.MatchingFields{"spec.nodeName": nodeName}
-		labelSelector := client.HasLabels{constants.DaemonSetRole}
-		kubeClient.EXPECT().List(ctx, gomock.Any(), labelSelector, fieldSelector)
+		labelSelector := client.HasLabels{constants.ModuleNameLabel}
+		kubeClient.EXPECT().List(ctx, gomock.Any(), labelSelector, fieldSelector).DoAndReturn(
+			func(_ interface{}, list *v1.PodList, _ ...interface{}) error {
+				list.Items = append(list.Items, pod1)
+				list.Items = append(list.Items, pod2)
+				return nil
+			},
+		)
 
-		_, err := helper.getDevicePluginPods(ctx, nodeName)
+		res, err := helper.getDevicePluginPods(ctx, nodeName)
 		Expect(err).ToNot(HaveOccurred())
+		Expect(len(res)).To(Equal(1))
+		Expect(res[0]).To(Equal(pod1))
 	})
 
 	It("error flow", func() {
 		fieldSelector := client.MatchingFields{"spec.nodeName": nodeName}
-		labelSelector := client.HasLabels{constants.DaemonSetRole}
+		labelSelector := client.HasLabels{constants.ModuleNameLabel}
```

Comment:
The worker pods also have that label. Aren't they going to interfere?

Reply:
That's why i have added the verification fro OwnerReference kind to DaemonSet